lib/fog/libvirt/models/compute/templates/server.xml.erb

```diff
@@ -46,7 +46,7 @@
     </console>
     <input type='tablet' bus='usb'/>
     <input type='mouse' bus='ps2'/>
-    <graphics type='<%= display[:type] %>' port='<%= display[:port] %>' autoport='yes' <% if display[:listen] and !(display[:listen].empty?)  %> listen='<%= display[:listen] %>'<% end %> <% if display[:password] and !(display[:password].empty?) %>passwd='<%=display[:password] %>'<% end %> />
+    <graphics type='<%= display[:type] %>' <% if display[:port] and !(display[:port].empty?) %>port='<%= display[:port] %>' autoport='no' <% else %>port='-1' autoport='yes' <%end%> <% if display[:listen] and !(display[:listen].empty?)  %> listen='<%= display[:listen] %>'<% end %> <% if display[:password] and !(display[:password].empty?) %>passwd='<%=display[:password] %>'<% end %> />
```

Comment:
What if we separate the definition of `port` and `autoport` in plain ruby before and in line 49 we just print its content? I believe this would make whats happening here more clear. What do you think?